apps/angular/src/core/services/anime-query-parameters.ts

```diff
@@ -0,0 +1,129 @@
+	private readonly router = inject(Router);
+
+	/** Anime query parameters. */
+	protected animeParameters: AnimeQueryParameters = {
+		offset: START_PAGE_INDEX,
+		limitPerPage: DEFAULT_PAGE_SIZE,
+		searchQuery: '',
+		animeType: [],
+		animeOrdering: [],
+	};
+
+	/** Returns an object with row query parameters. */
+	public getQueryParameters(): Observable<AnimeQueryParametersDto> {
+		return this.route.queryParams.pipe(
+			map(params => {
+				if (Object.keys(params).length === 0) {
+					this.navigate(this.animeParameters);
+				}
+				let parameters: AnimeQueryParametersDto = {
+					offset: +params['offset'] || START_PAGE_INDEX,
+					limit: +params['limit'] || DEFAULT_PAGE_SIZE,
+				};
+				if (params['search']) {
+					parameters = { ...parameters, search: params['search'] };
+				}
+				if (params['type__in']) {
+					// Disable eslint for a name of a dto parameter
+					// eslint-disable-next-line @typescript-eslint/naming-convention
+					parameters = { ...parameters, type__in: params['type__in'] };
+				}
+				if (params['ordering']) {
+					parameters = { ...parameters, ordering: params['ordering'] };
+				}
+				return parameters;
+			}),
+		);
+	}
+
+	/** Returns an object with parsed query parameters. */
+	public getParsedQueryParameters(): Observable<AnimeQueryParameters> {
+		return this.getQueryParameters().pipe(
+			switchMap(parameters => of(AnimeQueryParametersMapper.fromDto(parameters))),
+		);
+	}
+
+	/**
+	 * Changes query parameters for the current url.
+	 * @param animeQueryParameters Object with anime query parameters.
+	 */
+	private navigate(animeQueryParameters: AnimeQueryParameters): void {
+		const queryParams = { ...AnimeQueryParametersMapper.toDto(animeQueryParameters) };
+		this.router.navigate([], {
+			queryParams,
+		});
+	}
+
+	/**
+	 * Changes query parameters when pagination event occurs.
+	 * @param event Page change event.
+	 */
+	public onPageChange(event: PageEvent): void {
+		const offset = event.pageIndex * event.pageSize;
+		const limitPerPage = event.pageSize;
+		this.animeParameters = { ...this.animeParameters, offset, limitPerPage };
+		this.navigate(this.animeParameters);
+	}
+
+	/**
+	 * Changes query parameters when new type gets selected.
+	 * @param animeType An object with selected anime types.
+	 */
+	public onSelectType(animeType: AnimeType[]): void {
+		const offset = START_PAGE_INDEX;
+		this.animeParameters = { ...this.animeParameters, offset, animeType };
+		this.navigate(this.animeParameters);
+	}
+
+	/**
+	 * Changes query parameters when search event occurs.
+	 * @param searchQuery Search anime query.
+	 */
+	public onSearch(searchQuery: string): void {
+		const offset = START_PAGE_INDEX;
+		this.animeParameters = { ...this.animeParameters, offset, searchQuery };
+		this.navigate(this.animeParameters);
+	}
+
+	/**
+	 * Changes query parameters when sort event is triggered.
+	 * @param event Sort event from an anime table.
+	 */
+	public onSortChange(event: SortParameter): void {
```

Comment:
Please don't use preffix `on` in services. Is is not handler
```suggestion
	public chageSort(sort: SortParameter): void {
```